__tests__/app.test.js

```diff
@@ -318,7 +322,100 @@ describe('/api', () => {
+
+            const comment =
+            {
+                author: 'thebeebop',
+                body: 'If you are reading this then you are a... fine example of a human being!'
+            }
+
+           return request(app)
+           .post('/api/articles/10/comments')
+           .expect(201)
+           .send(comment)
+           .then((response) => {
+               expect(response.body).toEqual( 
+                {
+                   comment_id: expect.any(Number),
+                   body: 'If you are reading this then you are a... fine example of a human being!',
+                   author: 'thebeebop',
+                   article_id: 10,
+                   votes: expect.any(Number),
+                   created_at: expect.any(String)
+
+               })
+               
+
+           })
+         });
+         test('400: Given a malformed body, respond with a 400 bad request.', () => {
+
+            const badComment = {};
+            return request(app)
+            .post('/api/articles/10/comments')
+            .expect(400)
+            .send(badComment)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Bad Request')
+            })
+         })
+         test('400: Given an incorrect data-type in object, respond with a 400 bad request.', () => {
+
+            const badComment =
+            {
+                author: 56,
+                body: 'If you are reading this then you are a... fine example of a human being!'
+            }
+
+             return request(app)
+             .post('/api/articles/10/comments')
+             .expect(400)
+             .send(badComment)
+             .then((response) => {
+                 expect(response.body.msg).toEqual('Bad Request')
+
+             })
+         })
+         test('400: Given an invalid ID, respond with a 400 bad request.', () => {
+
+            const comment =
+            {
+                author: 'thebeebop',
+                body: 'If you are reading this then you are a... fine example of a human being!'
+            }
+
+             return request(app)
+             .post('/api/articles/mario/comments')
+             .expect(400)
+             .send(comment)
+             .then((response) => {
+                 expect(response.body.msg).toEqual('Bad Request')
+
+             })
+         })
+         test('400: Given a valid articleID that does not exist, respond with a 400 bad request message.', () => {
+
+            const comment =
+            {
+                author: 'thebeebop',
+                body: 'If you are reading this then you are a... fine example of a human being!'
+            }
+
+            return request(app)
+            .post('/api/articles/90/comments')
+            .expect(400)
+            .send(comment)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Bad Request')
+
+            })
+         })
+    });
 
```

Comment:
I would say this error is a 404 not found rather than 400 bad request

Reply:
One more test. What if the username is a valid data type but that user doesn't exist in the db?